cajita/src/Cajita_Array.hpp

```diff
@@ -471,31 +545,59 @@ void update( Array_t& a, const typename Array_t::value_type alpha,
         } );
 }
 
+template <class Array_t, class DecompositionTag>
+std::enable_if_t<2 == Array_t::num_space_dim, void>
+update( Array_t& a, const typename Array_t::value_type alpha, const Array_t& b,
+        const typename Array_t::value_type beta, DecompositionTag tag )
+{
+    static_assert( is_array<Array_t>::value, "Cajita::Array required" );
+    auto a_view = a.view();
+    auto b_view = b.view();
+    Kokkos::parallel_for(
+        "ArrayOp::update",
+        createExecutionPolicy( a.layout()->indexSpace( tag, Local() ),
+                               typename Array_t::execution_space() ),
+        KOKKOS_LAMBDA( const long i, const long j, const long l ) {
+            a_view( i, j, l ) =
+                alpha * a_view( i, j, l ) + beta * b_view( i, j, l );
+        } );
+}
+
 //---------------------------------------------------------------------------//
 // Dot product
-template <class ViewType>
+template <class ViewType, std::size_t NumSpaceDim>
```

Comment:
Should these structs have a default? Not sure if it makes sense to use directly in an application code

Reply:
I don't think a default is needed here - especially since `NumSpaceDim` can be deduced from the `Array` used to get the view to construct the functor.